bin/runtime/src/lib.rs

```diff
@@ -344,13 +345,89 @@ parameter_types! {
     pub const Offset: u32 = 0;
 }
 
+// Choose a subset of all the validators for current era that contains all the
+// reserved nodes. Non reserved ones are chosen in consecutive batches for every session
+fn rotate() -> Option<Vec<AccountId>> {
+    let current_era = match Staking::active_era() {
+        Some(ae) if ae.index > 0 => ae.index,
+        _ => return None,
+    };
+    let mut all_validators: Vec<AccountId> =
+        pallet_staking::ErasStakers::<Runtime>::iter_key_prefix(current_era).collect();
+
+    let mut validators = pallet_elections::ErasReserved::<Runtime>::get();
+    all_validators.retain(|v| !validators.contains(v));
+    let n_all_validators = all_validators.len();
+
+    let n_validators = MembersPerSession::get() as usize;
+    let free_seats = n_validators.checked_sub(validators.len()).unwrap();
+
+    // The validators for the committee at the session `n` are chosen as follow:
+    // 1. Reserved validators are always chosen.
+    // 2. Given non-reserved list of validators the chosen ones are from the range:
+    // `n * free_seats` to `(n + 1) * free_seats` where free_seats is equal to free number of free
+    // seats in the committee after reserved nodes are added.
+    let current_session = Session::current_index() as usize;
+    let first_validator = current_session * free_seats;
+
+    validators.extend(
+        (first_validator..first_validator + free_seats)
```

Comment:
Are we planning to add later some unit tests to `rotate`? Mainly for sake of the readability of any future reader. I know it's very simple algo, yet maybe readability can benefit from a simple UT which imitates runtime of `run_nodes.sh -v 6`? Or maybe e2e test would be a better choice here